examples/nips17_adversarial_competition/dataset/download_images_multiprocessed.py

```diff
@@ -0,0 +1,109 @@
+"""Script which downloads dataset images using multiple processes.
+
+Usage:
+  python download_images_multiprocessed.py --input_file=INPUT_FILE --output_dir=IMAGES_DIR
+
+where:
+  INPUT_FILE is input csv file with dataset description, i.e. dev_dataset.csv
+  IMAGES_DIR is output directory where all images should be downloaded
+
+Example:
+  # create directory for images
+  mkdir images
+  # download images declared in dev_dataset.csv using multiple processes
+  python download_images_multiprocessed.py --input_file=dev_dataset.csv --output_dir=images
+
+
+Dependencies:
+  Python 2.7 or higher.
+  Pillow library: https://python-pillow.org/
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import os
+import sys
+import csv
+from multiprocessing.dummy import Pool as ThreadPool
+from functools import partial
```

Comment:
If you don't use any non-standard libraries then it would be actually preferable to merge download_images.py and download_images_multiprocessed.py into single file. And control multiprocessing with command line flag.